sklearn/feature_selection/from_model.py

```diff
@@ -199,9 +215,33 @@ def _get_support_mask(self):
                 'Either fit the model before transform or set "prefit=True"'
                 ' while passing the fitted estimator to the constructor.')
         scores = _get_feature_importances(estimator)
-        self.threshold_ = _calculate_threshold(estimator, scores,
-                                               self.threshold)
-        return scores >= self.threshold_
+        mask = np.zeros(scores.shape, dtype=bool)
+        if self.threshold is not None and self.max_n_features is not None:
```

Comment:
A subtlety: for consistency with `SelectKBest` in case of ties, I'd rather take suffixes of `np.argsort(scores,...)` than prefixes of `np.argsort(-scores, ...)`.